pkg/rpcclient/waiter/waiter.go

```diff
@@ -244,15 +250,15 @@ func (w *PollingBased) WaitAny(ctx context.Context, vub uint32, hashes ...util.U
 // EventBased contains PollingBased under the hood and falls back to polling when subscription-based
 // awaiting fails.
 func NewEventBased(waiter RPCEventBased) (*EventBased, error) {
-	return NewCustomEventBased(waiter, PollConfig{})
+	return NewCustomEventBased(waiter, Config{})
 }
 
 // NewCustomEventBased creates an instance of Waiter supporting websocket event-based transaction awaiting.
 // EventBased contains PollingBased under the hood and falls back to polling when subscription-based
-// awaiting fails. PollingBased configuration options may be specified via pollConfig parameter
+// awaiting fails. Waiter configuration options may be specified via config parameter
 // (defaults are used if not specified).
-func NewCustomEventBased(waiter RPCEventBased, pollConfig PollConfig) (*EventBased, error) {
-	polling, err := NewCustomPollingBased(waiter, pollConfig)
+func NewCustomEventBased(waiter RPCEventBased, config Config) (*EventBased, error) {
```

Comment:
Also, `NewCustomPollingBased` still accepts `PollConfig`. Change it to accept `Config`?

Reply:
No, it's type-specific, so it should have type-specific options.